test_runner/fixtures/benchmark_fixture.py

```diff
@@ -23,6 +23,10 @@
 from fixtures.log_helper import log
 from fixtures.neon_fixtures import NeonPageserver
 
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from typing import Callable, ClassVar, Optional
```

Comment:
Is there some harm in always importing these? I.e. why do these need to be guarded by TYPE_CHECKING?

(comment applies to all such blocks in the PR)

Reply:
It's a micro-optimization to help tests at runtime. The Python parser will skip `TYPE_CHECKING` blocks.

Reply:
https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

Reply:
We could move a lot more, but it requires more developer discipline. I just got some low hanging fruit in relation to this PR. I wonder if there are tools to detect imports that could move to `TYPE_CHECKING` blocks.